x/twap/api_test.go

```diff
@@ -182,9 +236,106 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 				s.twapkeeper.StoreNewRecord(s.Ctx, record)
 			}
 			s.Ctx = s.Ctx.WithBlockTime(test.ctxTime)
-			twap, err := s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
+
+			var twap sdk.Dec
+			var err error
+
+			twap, err = s.twapkeeper.GetArithmeticTwap(s.Ctx, test.input.poolId,
 				test.input.quoteAssetDenom, test.input.baseAssetDenom,
 				test.input.startTime, test.input.endTime)
+
+			if test.expErrorStr != "" {
+				s.Require().Error(err)
+				s.Require().Contains(err.Error(), test.expErrorStr)
+				return
+			}
+			s.Require().NoError(err)
+			s.Require().Equal(test.expTwap, twap)
+		})
+	}
+}
+
+// TestGetArithmeticTwap tests if we get the expected twap value from `GetArithmeticTwap`.
+// We test the method directly by updating the accumulator and storing the twap records
+// manually in this test.
+// This test also includes testing the method `TestGetArithmeticTwapToNow`.
+func (s *TestSuite) TestGetArithmeticTwapToNow() {
+	type getTwapInput struct {
```

Comment:
Wdyt about defining this at the top of the file to be shared between the tests?

Reply:
good call!